CI: make test-gen-api-docs is failing quite often

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
If the matter is security related, please disclose it privately via 
-->


**What happened**:
Multiple Builds were failed:
https://console.cloud.google.com/cloud-build/builds/74de68e2-12de-4ab5-b8e1-ed0c4b6f47b0;step=17?project=agones-images
in this Pull Request for instance: https://github.com/googleforgames/agones/pull/1685
```
< <h3 id="multicluster.agones.dev/v1.GameServerAllocationPolicy">GameServerAllocationPolicy
< <h3 id="multicluster.agones.dev/v1.GameServerAllocationPolicySpec">GameServerAllocationPolicySpec
3260a3266
> <p>
5211c5217
< This field is beta-level and is enabled by default, could be disabled by the &ldquo;ContainerPortAllocation&rdquo; feature.</p>
---
> This field is alpha-level and is only honored by servers that enable the &ldquo;ContainerPortAllocation&rdquo; feature.</p>
5517,5518c5523,5524
< {{% feature expiryVersion="1.8.0" %}}
< {{% feature publishVersion="1.8.0" %}}
---
> {{% feature expiryVersion="1.7.0" %}}
> {{% feature publishVersion="1.7.0" %}}
includes/website.mk:98: recipe for target 'test-gen-api-docs' failed
```
**What you expected to happen**:
No fails on `make test-gen-api-docs`

**How to reproduce it (as minimally and precisely as possible)**:
Could not reproduce as of know.

**Anything else we need to know?**:

**Environment**:
- Agones version: master (1.8.0-dev)
- Kubernetes version (use `kubectl version`): n/a
- Cloud provider or hardware configuration: CloudBuilds
- Install method (yaml/helm): 
- Troubleshooting guide log(s):
- Others:

--- a/site/gen-api-docs.sh
+++ b/site/gen-api-docs.sh
@@ -62,7 +62,7 @@ function sedeasy {
 }
 
 # do we have changes in generated API docs compared to previous version
-if ! diff $RESULT $OLD;
+if ! diff <(sort $RESULT) <(sort $OLD);
 then 
   echo "Output to a file $FILE"
 

